### PR TITLE
move label_smmoth XPU kernel to PHI,test=kunlun

### PR DESCRIPTION
**Pull request by risemeup1**
### PR types
others

### PR changes
others

### Describe
move label_smmoth XPU kernel to PHI

### CODE REVIEW

File: paddle/phi/kernels/xpu/label_smooth_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+template <typename T, typename Context>
+void LabelSmoothKernel(const Context& ctx,
+                       const DenseTensor& label,
+                       const paddle::optional<DenseTensor>& prior_dist,
+                       float epsilon,
+                       DenseTensor* out) {
+  auto label_dim = label.dims()[label.dims().size() - 1];
+  auto ptr = ctx.template Alloc<T>(out);
+  if (prior_dist) {
```
Comment (zhangbo9674) [inline paddle/phi/kernels/xpu/label_smooth_kernel.cc]: 这里的逻辑是判断optional类型参数prior_dist是否有初始化吧？可以通过prior_dist.is_initialized()方法来判断

Comment (risemeup1) [reply paddle/phi/kernels/xpu/label_smooth_kernel.cc]: 已收到并按要求修改，感谢指正🙏

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Review by ykkk2333 — APPROVED** [review]
lgtm

**Review by ykkk2333 — APPROVED** [review]
LGTM

**Review by YuanRisheng — APPROVED** [review]

**Pull request merged**